### PR TITLE
Hooked up slack integration

### PR DESCRIPTION
**Pull request by tpcstld**
Config file:
```
etl:
  slack:
    api_token:
    channel_name:
    [bot_username:]
```
`bot_username` is optional, defaults to 'Dataduct'

### CODE REVIEW

File: dataduct/utils/slack_hook.py [hunk 1]
```diff
@@ -13,21 +13,33 @@ def post_message(message):
...
+    # If any of these fail, silently skip because the user doesn't know about
+    # the slack integration or doesn't care
     try:
         import slack
         import slack.chat
         config = Config()
         slack_config = config.etl['slack']
+    except KeyError:
+        return
+
```
Comment (sb2nov) [inline dataduct/utils/slack_hook.py]: you can just do:

Do this before the slack imports:
```
slack_config = config.get('slack', None)
if slack_config is None: 
     return
```

File: dataduct/utils/slack_hook.py [hunk 2]
```diff
@@ -13,21 +13,33 @@ def post_message(message):
...
-        logger.info('1) Run: pip install pyslack')
-        logger.info('2) Visit https://api.slack/com/web to generate a token')
-        logger.info('   and add:')
-        logger.info('       api_token:<your token>')
-        logger.info('       channel_name:<channel_name>')
-        logger.info('       bot_username:<bot_username>')
-        logger.info('   to the etl section of your config file')
+                                username=slack_config.get('bot_username',
+                                                          'Dataduct'))
+    except KeyError:
```
Comment (sb2nov) [inline dataduct/utils/slack_hook.py]: you want to catch import error on slack as that might not be installed after the above change. 
So you may still want to catch a generic exception as posting on slack should never ever block you from running pipelines.

File: dataduct/utils/slack_hook.py [hunk 3]
```diff
@@ -13,21 +13,33 @@ def post_message(message):
...
+                   '1) Run: pip install pyslack',
+                   '2) Visit https://api.slack/com/web to generate a token',
+                   '3) Add:',
+                   '       api_token:<your token>',
+                   '       channel_name:<channel_name>',
+                   '       bot_username:<bot_username>',
+                   '   to the etl section of your config file']
+        for line in message:
+            logger.error(line)
+        raise
```
Comment (sb2nov) [inline dataduct/utils/slack_hook.py]: don't raise

File: dataduct/etl/etl_actions.py [hunk 4]
```diff
@@ -84,6 +85,8 @@ def activate_pipeline(etl):
     logger.info('Activated pipeline. Id: %s', etl.pipeline.id)
     logger.info('Monitor pipeline here: %s',
                 URL_TEMPLATE.format(ID=etl.pipeline.id))
+    # Post a slack message
+    post_message('Pipeline started: %s' % etl.name)
```
Comment (sb2nov) [inline dataduct/etl/etl_actions.py]: Should we record who started it somehow ? Is there a way ?

**Comment by sb2nov** [comment]
LGTM

**Comment by sungjuly** [comment]
LGTM +1

Do you have a plan to add hipchat?

**Comment by sb2nov** [comment]
@sungjuly we don't have an account so it is hard to test hipchat. Feel free to take a stab at it, we can just change the post message function to try out things based on some config.

**Pull request merged**